kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java

```diff
@@ -272,12 +264,8 @@ private void updateTopics(String deploymentId, ProcessDefinition processDefiniti
     }
 
     private void removeTopics(String deploymentId, ProcessDefinition processDefinition) {
-        if (processSignals()) {
-            removeTopics(topic2Signal, deploymentId, processDefinition.getSignalsDesc());
-        }
-        if (processMessages()) {
-            removeTopics(topic2Message, deploymentId, processDefinition.getMessagesDesc());
-        }
+        removeTopics(topic2Signal, deploymentId, processDefinition.getSignalsDesc());
```

Comment:
This change tries to cover an edge case. System property enable when onDeploy/onActivate is called, but disable when onUndeploy/onDeactivate is invoked. This will  to a subscripton "leak". In order to avoid that, at the cost of sacrificing a bit of performance (not much in any case), it is better to clean always

Reply:
OK, agreed; it's very weird but not impossible. Anyway, performance impact is minimum and removing topics is during undeployment.